tests/FakeItEasy.Tests/Creation/CastleDynamicProxy/CastleDynamicProxyInterceptionValidatorTests.cs

```diff
@@ -42,7 +42,9 @@ public CastleDynamicProxyInterceptionValidatorTests()
                     () => object.Equals("foo", "bar"),
                     "Static methods can not be intercepted."),
                 new NonInterceptableTestCase(
-                    () => "foo".Count(),
+                #pragma warning disable CA1806 // Do not ignore method results
+                    () => "foo".Distinct(),
+                #pragma warning restore CA1806 // Do not ignore method results

```

Comment:
I don't understand this change... Why replace Count with Distinct?

Reply:
Because `Count()` gives a warning about us not using `Length`. And a similar warning about us not using `Length`, but slightly different AND CA1806. I preferred one pragma to 3!

Reply:
I should've said.

Reply:
Ah, I see. Probably doesn't matter here, but `Distinct()` has no effect unless the result is enumerated, so it looked a bit strange

Reply:
Yeah, we just need to identify a method. I can comment up, if you like.

Reply:
No need. Thanks for the explanation